twindb_backup/source/mysql_source.py

```diff
@@ -1,32 +1,101 @@
+# -*- coding: utf-8 -*-
+"""
+Module defines MySQL source class for backing up local MySQL.
+"""
 import os
-import shlex
 import tempfile
 import time
 
 from ConfigParser import NoOptionError
 from contextlib import contextmanager
-from pymysql.err import InternalError
 from subprocess import Popen, PIPE
-from twindb_backup import log, get_files_to_delete
+
+import pymysql
+
+from twindb_backup import LOG, get_files_to_delete, INTERVALS
 from twindb_backup.source.base_source import BaseSource
-from twindb_backup.util import get_connection
 
 
 class MySQLSourceError(Exception):
     """Errors during backups"""
 
 
+class MySQLConnectInfo(object):  # pylint: disable=too-few-public-methods
+    """MySQL connection details """
+    def __init__(self, defaults_file,
+                 connect_timeout=10,
+                 cursor=pymysql.cursors.DictCursor):
+
+        self.cursor = cursor
+        self.connect_timeout = connect_timeout
+        self.defaults_file = defaults_file
+
+
 class MySQLSource(BaseSource):
-    def __init__(self, defaults_file, run_type, config, dst):
-        self.defaults = defaults_file
-        self._suffix = 'xbstream.gz'
+    """MySQLSource class"""
+    def __init__(self, mysql_connect_info, run_type, full_backup, dst):
+        """
+        MySQLSource constructor
+
+        :param mysql_connect_info: MySQL connection details
+        :type mysql_connect_info: MySQLConnectInfo
+        :param run_type:
+        :param full_backup: When to do full backup e.g. daily, weekly
+        :type full_backup: str
+        :param dst:
+        """
+
+        class _BackupInfo(object):  # pylint: disable=too-few-public-methods
```

Comment:
BackupInfo should describe a status entry. At this moment it's not used outside init(), but you're right, we should check where else it may be used.
```
"master1/hourly/mysql/mysql-2017-02-13_04_58_53.xbstream.gz": {
            "binlog": "mysql-bin.000001",
            "config": [
                {
                    "/etc/my.cnf": "W215c3FsZF0KZGF0YWRpcj0vdmFyL2xpYi9teXNxbApzb2NrZXQ9L3Zhci9saWIvbXlzcWwvbXlzcWwuc29jawp1c2VyPW15c3FsCiMgRGlzYWJsaW5nIHN5bWJvbGljLWxpbmtzIGlzIHJlY29tbWVuZGVkIHRvIHByZXZlbnQgYXNzb3J0ZWQgc2VjdXJpdHkgcmlza3MKc3ltYm9saWMtbGlua3M9MAoKc2VydmVyX2lkPTEwMApndGlkX21vZGU9T04KbG9nLWJpbj1teXNxbC1iaW4KbG9nLXNsYXZlLXVwZGF0ZXMKZW5mb3JjZS1ndGlkLWNvbnNpc3RlbmN5CgpbbXlzcWxkX3NhZmVdCmxvZy1lcnJvcj0vdmFyL2xvZy9teXNxbGQubG9nCnBpZC1maWxlPS92YXIvcnVuL215c3FsZC9teXNxbGQucGlkCg=="
                }
            ],
            "lsn": 19621635,
            "position": 40713,
            "type": "full"
        }
```